test/forcing/TrivialForcingProvider.hpp

```diff
@@ -3,45 +3,49 @@
 
 #include <string>
 #include <vector>
-#include "ForcingProvider.hpp"
+#include "GenericDataProvider.hpp"
+
 
 #define OUTPUT_NAME_1 "output_name_1"
 #define OUTPUT_VALUE_1 42.0
 #define OUTPUT_DEFAULT_1 36.8
 
 using namespace std;
 
-namespace forcing {
+namespace data_access {
     namespace test {
 /**
  * A trivial implementation, strictly for testing OptionalWrappedProvider.
  */
-        class TrivialForcingProvider : public ForcingProvider {
+        class TrivialForcingProvider : public GenericDataProvider {
         public:
 
             TrivialForcingProvider() {
                 outputs.push_back(OUTPUT_NAME_1);
             }
-
-            const vector<string> &get_available_forcing_outputs() override {
+                                  
+            const std::vector<std::string>& get_avaliable_variable_names() override {
                 return outputs;
             }
 
-            time_t get_forcing_output_time_begin(const string &output_name) override {
+            long get_data_start_time() override {
                 return 0;
             }
 
-            time_t get_forcing_output_time_end(const string &output_name) override {
+            long get_data_stop_time() override {
+                return 1000000;
+            }
+
+            long record_duration() override {
                 return 1000000;
```

Comment:
Since this is `get_data_stop_time()` - `get_data_start_time()` this looks like another non-stride implementation of `record_duration()`...Though per the comments elsewhere, I'm not sure if 0, 1, min long, or another value is appropriate.

Reply:
zero would probably be a bad value